src/main/java/com/google/sps/servlets/CasesDataServlet.java

```diff
@@ -76,25 +80,100 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(reportsJson);
   }
 
-  /**
-   * Returns Json string of all reports
-   */
   public String getReportsJson() {
     return reportsJson;
   }
 
   /**
-   * Represents active number of cases at a specific lat lng point
+   * Establish connection to live Coivd-19 data set
+   */
+  private Scanner connectToData() {
+    URL covidDataUrl = null;
+    HttpURLConnection connection = null;
+    int responseCode = 0;
+    Scanner scanner = null;
+
+    try {
+      String date = getDate();
+      String url =
+          "https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_daily_reports/"
+          + date + ".csv";
+      covidDataUrl = new URL(url);
+      connection = (HttpURLConnection) covidDataUrl.openConnection();
+      connection.setRequestMethod("GET");
+      responseCode = connection.getResponseCode();
+      if (responseCode == HttpURLConnection.HTTP_OK) {
+        scanner = new Scanner(covidDataUrl.openStream());
+      }
+    } catch (IOException e) {
+      System.out.println("Unable to open connection");
+    }
+    return scanner;
+  }
+
+  /**
+   * Returns the date in mm-dd-yyyy format to retrieve most recent data set
+   */
+  private String getDate() {
+    Clock today = Clock.systemUTC();
+    Instant in = today.instant();
+    int hour = Integer.parseInt(in.toString().substring(11, 13));
+    in = in.minus(1, ChronoUnit.DAYS);
+    // The dataset updates at 5. If it is before 5, need previous set
+    if (hour < 5) {
+      in = in.minus(1, ChronoUnit.DAYS);
+    }
+    String dateTime = in.toString();
+    String year = dateTime.substring(0, 4);
+    String month = dateTime.substring(5, 7);
+    String day = dateTime.substring(8, 10);
+    return month + "-" + day + "-" + year;
```

Comment:
Same thing here, although maybe it would be better to just comment what the format of dateTime is, and let the reader see (i.e. // date format : yyyy/mm/dd)